### ISSUE TITLE
Context namespace does not exist

### ISSUE THREAD

**Issue by ubamrein** [w3c/cid#136]
In this [commit](https://github.com/w3c/cid/commit/75dd25051e8ee197d9ab3b6a3c8239a3f95c5e6c) the context string was changed from "../controller/.." to "../cid/..", but the document is not available under the new [URL](https://www.w3.org/ns/cid/v1) (so all examples provided fail) (and also not under the old one). Funny enough, the controller seems to be defined under the https://www.w3.org/ns/did/v1 namespace (according to https://w3c.github.io/vc-data-integrity/vocab/security/vocabulary.html#controller).

**Comment by msporny** [comment]
Yes, good catch. The current context URL is broken because we're in the middle of a specification rename and publication, which has gotten caught up in a month+ long administrative process. The URL will eventually resolve, and when it does, we'll note that in this issue.

**Comment by ubamrein** [comment]
Is there any preferred way, how to handle such cases (renaming/removing those context urls) when relaying on Json-LD expansion? Is it best to have local versions of the relevant documents?

**Comment by dlongley** [comment]
@ubamrein,

> Is there any preferred way, how to handle such cases (renaming/removing those context urls) when relaying on Json-LD expansion? Is it best to have local versions of the relevant documents?

Yes, for well-defined and eventually immutable contexts (such as those in this specification and others produced by the VCWG), local copies of the contexts should absolutely be installed in software (similar to library "imports") and treated as if "already resolved" (i.e., always use a JSON-LD document loader that loads from your local copy) when performing JSON-LD API operations such as expansion.

Note that once the specs are finalized, the contexts identified by specific URLs in the specifications are expected to be immutable such that local copies made at that point should never change.

**Comment by ubamrein** [comment]
Thanks for clarifying!

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2025-01-08-vcwg#section3-5) on 2025-01-08

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 3.5. Context namespace does not exist (issue cid#136)

_See github issue [cid#136](https://github.com/w3c/cid/issues/136)._




**Brent Zundel:** Context namespace does not exist. This is raised by and assigned to Ivan, what are next steps, Ivan?  

**Manu Sporny:** We just changed the name and the examples aren't working -- so we need to publish.  

**Ivan Herman:** We have a bunch of links that are temporary/wrong because we haven't registered the shortname, so until we are in CR these links are broken, they will get fixed when CR happens.  

**Manu Sporny:** +1 to that Ivan, so all the terminology references are broken too -- I can't merge things on DID core for example. This is a lot of broken links that will be resolved when we go to CR, but it's causing issues in the meantime. I don't know if there's any reason why the `ns` stuff ... it doesn't have to do with publishing to TR...  

**Ivan Herman:** It's the publishing of the document that is the source of the problems.  

**Manu Sporny:** It's just the URL that's being published and the name of the github version of the document path changed. If that's updated then this issue is resolved, which doesn't depend on TR space, I don't think.  

**Ivan Herman:** I have to check.  

**Manu Sporny:** I think it's just changing a URL in the `ns` space.  

**Ivan Herman:** We have a more generic problem and I'll look at this.  
… Apart from that, we said we'll do this official name change with CR and we could a working draft change and go through that and get this over with. It depends whether we want to do that as a separate process, which I'm not keen to do, but we could do that as well.  

**Brent Zundel:** I don't think that's necessary.  

**Ivan Herman:** Ok, good.  

**Brent Zundel:** For #136, we're going to see what's still broken after we go to CR, right?  

**Ivan Herman:** Yes.  

**Manu Sporny:** Yes.  

**Brent Zundel:** Ok, that's the CID issues.  
… Moving to bitstring status list.  
… Let's look at the open PRs first.  

</details>

**Comment by iherman** [comment]
The original issue has been solved now: https://www.w3.org/ns/cid/v1 works; a redirect has been set up to the new repository. For bacwkards compatibility, https://www.w3.org/ns/controller/v1 also works, though is now an unofficial URL for the context.

When moving to PR, the redirection should be removed in favor of placing a copy into /ns.

cc @msporny @brentzundel

**Comment by msporny** [comment]
> The original issue has been solved now: https://www.w3.org/ns/cid/v1 works; a redirect has been set up to the new repository. For backwards compatibility, https://www.w3.org/ns/controller/v1 also works, though is now an unofficial URL for the context.

Excellent, thank you @iherman!
